pkg/api/v20160930/validate.go

```diff
@@ -48,6 +48,11 @@ func (a *AgentPoolProfile) Validate() error {
 	if e := validateName(a.VMSize, "AgentPoolProfile.VMSize"); e != nil {
 		return e
 	}
+	if orchestratorType == Kubernetes {
```

Comment:
We need to do case insensitive comparison. See #529. You can check in first and I'll change.